Gems/GraphCanvas/Code/Source/Components/Nodes/General/GeneralNodeLayoutComponent.cpp

```diff
@@ -97,6 +100,18 @@ namespace GraphCanvas
     
     void GeneralNodeLayoutComponent::OnStyleChanged()
     {
+        if (!m_hasSwitchedToHorizontalLayout)
+        {
+            Styling::StyleHelper style(GetEntityId());
+            Qt::Orientation layoutOrientation = style.GetAttribute(Styling::Attribute::LayoutOrientation, Qt::Vertical);
```

Comment:
Is this new member bool necessary? Can you query the current LayoutOrientation on demand?